File: server/KomputerBudowanieAPI/KomputerBudowanieAPI/Models/PcConfiguration.cs
using System.ComponentModel.DataAnnotations;

namespace KomputerBudowanieAPI.Models
{
    public class PcConfiguration
    {

        [Key]
        public Guid Id { get; set; }

        public string? Name { get; set; }
        public string? Description { get; set; }

        public Motherboard? Motherboard { get; set; }
        public GraphicCard? GraphicCard { get; set; }
        public Cpu? Cpu { get; set; }
        public CpuCooling? CpuCooling { get; set; } = null;

        public Case? Case { get; set; }
        public PowerSupply? PowerSupply { get; set; } = null;
        public WaterCooling? WaterCooling { get; set; }
        public User? User { get; set; } = null;
        //public double TotalPrice { get; set; } = 0;
        public ICollection<PcConfigurationStorage>? PcConfigurationStorages { get; set; }
        public ICollection<PcConfigurationRam>? PcConfigurationRams { get; set; }
    }
}

File: client/src/models/PcConfiguration.js
import {
  Case,
  Cpu,
  CpuCooling,
  GraphicCard,
  Motherboard,
  PowerSupply,
  Ram,
  Storage,
  WaterCooling,
} from "./index";

const PcConfiguration = {
  case: Case,
  cpu: Cpu,
  cpuCooling: CpuCooling,
  graphicCard: GraphicCard,
  motherboard: Motherboard,
  powerSupply: PowerSupply,
  ram: [Ram],
  storage: [Storage],
  waterCooling: WaterCooling,
};

export default PcConfiguration;
